### ISSUE TITLE
False positive no-member on PonyORM

### ISSUE THREAD

**Issue by fluffy-critter** [PyCQA/pylint#4690]
### Steps to reproduce

On pylint (since 2.9.3, and still experiencing it on 3.0.0a4) I am having a strange case where pylint is generating a `no-member` error for a PonyORM model class, but only on one line of code.

The code in question is at https://github.com/PlaidWeb/Publ/blob/ee40cfcd388ee31b3e5c977b5ca326ce0af1e013/publ/entry.py#L841 and is excerpted below:

```
    # Update the entry ID
    if str(record.id) != entry['Entry-ID']:   # here, pylint complains about record.id
        del entry['Entry-ID']
        entry['Entry-ID'] = str(record.id)   # but pylint doesn't complain here
        fixup_needed = True
```

I also use the `.id` member in many other places in my code without issue (for example, line 55 of that same file).

### pylint --version output

Result of `pylint --version` output:

```
pylint 2.9.3
astroid 2.6.2
Python 3.8.6 (default, Jan  6 2021, 12:14:27) 
[Clang 12.0.0 (clang-1200.0.32.28)]
```
